### PR TITLE
Remove compatibility convert for output

### PR DESCRIPTION
**Pull request by nitbharambe**

### CODE REVIEW

File: src/power_grid_model/core/data_handling.py [hunk 1]
```diff
@@ -113,13 +116,22 @@ def create_output_data(
...
         else:
             shape = (count,)
-        result_dict[name] = initialize_array(output_type, name, shape=shape, empty=True)
+
+        requested_component = processed_output_types[name]
+        dtype = power_grid_meta_data[output_type][name].dtype
+        if requested_component is None:
+            result_dict[name] = initialize_array(output_type, name, shape=shape, empty=True)
+        elif isinstance(requested_component, ComponentAttributeFilterOptions):
+            result_dict[name] = {attr: np.empty(shape, dtype=dtype[attr]) for attr in cast(Iterable[str], dtype.names)}
```
Comment (mgovers) [inline src/power_grid_model/core/data_handling.py:133]: why do you need to cast?

**Review by figueroa1395 — COMMENTED** [review]
Looks good to me, I just have very minor comments. Also, nice testing.

File: tests/unit/test_data_handling.py [hunk 2]
```diff
@@ -12,57 +13,121 @@
...
+    """Test output_data creation. Batch size is set later in the test."""
     all_component_count = {CT.node: 4, CT.sym_load: 3, CT.source: 1}
-    batch_size = 15 if is_batch else 1
     actual = create_output_data(
         output_component_types=output_component_types,
         output_type=DT.sym_output,
         all_component_count=all_component_count,
-        is_batch=is_batch,
-        batch_size=batch_size,
+        is_batch=False if batch_size == () else True,
```
Comment (mgovers) [inline tests/unit/test_data_handling.py:126]:
```suggestion
        is_batch=bool(batch_size),
```

File: tests/unit/test_data_handling.py [hunk 3]
```diff
@@ -12,57 +13,121 @@
...
+        batch_size=1 if batch_size == () else batch_size[0],
```
Comment (mgovers) [inline tests/unit/test_data_handling.py:127]: really? shouldn't it be `0`?

File: tests/unit/test_data_handling.py [hunk 4]
```diff
@@ -12,57 +13,121 @@
...
+        for attr in component_dtype.names
+        if attr in attributes
+    }
+
+
+def row_array(component_type, n_components, batch_size_tuple=()):
+    return initialize_array(DT.sym_output, component_type, (n_components,) + batch_size_tuple)
+
+
+@pytest.fixture(params=[(), (15,)])
```
[the remark this answers is not in the document]
Comment (mgovers) [reply tests/unit/test_data_handling.py:34]: see my comment below: add as kwargs

[the remark this answers is not in the document]
Comment (nitbharambe) [reply tests/unit/test_data_handling.py:34]: Yes indeed, empty batch as in single dataset.
I  keep it as tuple for easy appending to the `shape`

File: src/power_grid_model/core/power_grid_model.py [hunk 5]
```diff
@@ -596,7 +606,7 @@ def calculate_short_circuit(
         calculation_method: CalculationMethod | str = CalculationMethod.iec60909,
         update_data: Optional[dict[str, np.ndarray | dict[str, np.ndarray]] | Dataset] = None,
         threading: int = -1,
-        output_component_types: Optional[set[ComponentTypeVar] | list[ComponentTypeVar]] = None,
+        output_component_types: ComponentAttributeMapping = None,
```
Comment (mgovers) [inline src/power_grid_model/core/power_grid_model.py:609]: please update docstring

**Review by mgovers — APPROVED** [review]

**Pull request merged**